packages/react-native-audio-api/common/cpp/audioapi/core/AudioNode.cpp

```diff
@@ -1,15 +1,35 @@
-#include <audioapi/core/AudioNode.h>
+#include "AudioNode.h"
 #include <audioapi/core/AudioParam.h>
 #include <audioapi/core/BaseAudioContext.h>
 #include <audioapi/core/utils/AudioNodeManager.h>
-#include <audioapi/utils/AudioArray.h>
+#include <audioapi/core/utils/NodeConnections.h>
 #include <audioapi/utils/AudioBus.h>
+#include <stdexcept>
 
 namespace audioapi {
 
 AudioNode::AudioNode(BaseAudioContext *context) : context_(context) {
-  audioBus_ = std::make_shared<AudioBus>(
-      RENDER_QUANTUM_SIZE, channelCount_, context->getSampleRate());
+  m_connections = std::make_unique<NodeConnections>(this, context);
+  isInitialized_ = true;
+
+  // Initialize output buses for a standard node
+  m_outputBuses.resize(numberOfOutputs_);
+  for (unsigned int i = 0; i < numberOfOutputs_; ++i) {
+    m_outputBuses[i] = std::make_shared<AudioBus>(
+        RENDER_QUANTUM_SIZE, channelCount_, context->getSampleRate());
+  }
+}
```

Comment:
```suggestion
AudioNode::AudioNode(BaseAudioContext *context) : AudioNode(context, numberOfInputs_) {}
```
you can use delegate constructor here

Reply:
switched to just one `AudioNode::AudioNode(BaseAudioContext *context, unsigned int numberOfInputs)` constructor with the `numberOfInputs` defaulting to 1